src/lib/alloc.c

```diff
@@ -945,7 +942,8 @@ static void *_balloc_unlocked(uint32_t flags, uint32_t caps, size_t bytes,
 		/* Continue from the next heap */
 	}
 
-	return ptr;
+	return (flags & SOF_MEM_FLAG_COHERENT) && (CONFIG_CORE_COUNT > 1) ?
```

Comment:
if SOF_MEM_FLAG_COHERENT == 1, then we prefer uncache addr ?  the flag name is confusing. Or named SOF_MEM_FLAG_UNCACHE